facturx/tests/test_facturx.py

```diff
@@ -1,23 +1,86 @@
 import os
 import unittest
-from facturx.facturx import FacturX
+from facturx.facturx import *
+from lxml import etree
+
 
 class TestReading(unittest.TestCase):
-    def setUp(self):
+    def discover_files(self):
         self.test_files_dir = os.path.join(os.path.dirname(__file__), 'sample_invoices')
         self.test_files = os.listdir(self.test_files_dir)
 
     def test_from_file(self):
+        self.discover_files()
         for file in self.test_files:
             file_path = os.path.join(self.test_files_dir, file)
             FacturX(file_path)
 
-    def test_input_error(self):
-        with self.assertRaises(TypeError) as context:
-            FacturX('non-existant.pdf')
+    # returning file path for a specific file in 'sample_invoices'
+    def find_file(self, file_name):
+        self.discover_files()
+        for file in self.test_files:
+            if file == file_name:
+                file_path = os.path.join(self.test_files_dir, file)
+                return file_path
+
+    # def test_input_error(self):
+    #     with self.assertRaises(TypeError) as context:
+    #         FacturX('non-existant.pdf')
+
+    def test_file_without_embedded_data(self):
+        file_path = self.find_file('no_embedded_data.pdf')
+        self.assertEqual(FacturX(file_path)._xml_from_file(file_path), None)
+
+    def test_file_embedded_data(self, file_name='embedded_data.pdf'):
+        file_path = self.find_file(file_name)
+        if FacturX(file_path)._xml_from_file(file_path) is None:
```

Comment:
This needs to be rewritten to use `assertTrue` directly. The `if`-condition is useless.

Reply:
Same thing again below.